src/showyourwork/cli/commands/setup.py

```diff
@@ -47,6 +50,59 @@ def callback(code, stdout, stderr):
 
     name = f"@{user}".replace("_", "")
 
+    # Fetch latest showyourwork-action version if not specified
+    def get_latest_action_version():
+        # Get latest release info
+        release_url = "https://api.github.com/repos/showyourwork/showyourwork-action/releases/latest"
+        main_url = (
+            "https://api.github.com/repos/showyourwork/showyourwork-action/commits/main"
+        )
+        compare_url = "https://api.github.com/repos/showyourwork/showyourwork-action/compare/{release_tag}...main"
+        try:
+            release_resp = requests.get(release_url, timeout=5)
+            if release_resp.status_code == 200:
+                release_data = release_resp.json()
+                release_tag = release_data.get("tag_name")
+                # release_sha is not needed for comparison logic
+                # Get latest commit on main
+                main_resp = requests.get(main_url, timeout=5)
+                if main_resp.status_code == 200:
+                    main_data = main_resp.json()
+                    main_sha = main_data.get("sha")
+                    # Compare release tag with main
+                    if release_tag and main_sha:
+                        # Check if main has new commits since release
+                        comp_url = compare_url.format(release_tag=release_tag)
+                        comp_resp = requests.get(comp_url, timeout=5)
+                        if comp_resp.status_code == 200:
+                            comp_data = comp_resp.json()
+                            ahead_by = comp_data.get("ahead_by", 0)
+                            if ahead_by > 0:
+                                return main_sha
+                        # If no new commits, use release tag
+                        return release_tag
+        except Exception:
+            pass
+        # Fallback: just get latest commit SHA from main
+        try:
+            main_resp = requests.get(main_url, timeout=5)
+            if main_resp.status_code == 200:
+                main_data = main_resp.json()
+                main_sha = main_data.get("sha")
+                if main_sha:
+                    return main_sha
+        except Exception:
+            pass
+        # Final fallback
+        return "v1"
+
+    if not action_version:
+        action_version = get_latest_action_version()
+
+    # Set action_spec to local showyourwork version if not provided
+    if not action_spec:
+        action_spec = __version__
```

Comment:
Actually I see an issue here which might be very wrong

this should not be the version but something like 

git+https://github.com/showyourwork/showyourwork

so right now this will make any project remote fail

Reply:
because since I had to test this specific branch (of the PR) when I created the project I did it passing the spec with the `-a` option as

`showyourwork setup -a git+https://github.com/showyourwork/showyourwork@fix-cookiecutter_template_versions foo/bar`

and so this conditional statement never went through

(which is actually a missing remote test)

before I add the test I need to figure out if we are covering all the cases: the obvious solution is to set 

`git+https://github.com/showyourwork/showyourwork`

am I missing something?

Reply:
OK I fixed it, see below in the thread for a detailed explanation